CONTRIBUTING.md

```diff
@@ -1,3 +1,192 @@
 # Contributing
-
 This repo is intended for internal (Snyk) contributions only at this time.
+
+# Creating a new extension
+The `go-application-framework` makes it easy to add functionality to the Snyk CLI by taking care of the orchestration, requirements, and state management when running the CLI; this all happens behind the scenes via the packages provided by the framework. It means we can focus on building the core business logic for our extension and not worry about much else.
+
+Creating a new extension using the `go-application-framework` can be done in three steps
+
+1. Create the extension workflow
+2. Register the extension with the workflow engine
+3. Initialise the extension when using the `go-application-framework`
+
+# Create the extension workflow
+The extension workflow is the main component of your extension, it should:
+1. Uniquely identify your workflow
+2. Initialise your workflow (registering the extension with the workflow engine should happen here)
+1. Have an entry point containing the extension business logic
+
+# Creating `snyk whoami`
+The `snyk whoami` command was created following the steps outlined above; the following outlines the extension implementation in detail.
+
+## Uniquely identify your workflow
+Before we can start creating our extension workflow, we'll need to create our workflow identifier. An identifier is required by the workflow engine to identify our workflow. The workflow identifier should also contain the name of the new extension command.
+
+The framework's `workflow` package allows us to easily create an identifier in this format as follows:
+
+```go
+import "github.com/snyk/go-application-framework/pkg/workflow"
+
+// define a new workflow identifier for this workflow
+// this identifies the 'snyk whoami' command with the workflow engine
+var workflowName = "whoami"
```

Comment:
maybe give the example as `snyk.whoami`, as we want external contributions, too :)